src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacLLMFreeTextValidator.java

```diff
@@ -0,0 +1,253 @@
+
+    private final AbstractConfigLoader configLoader;
+    private final ObjectMapper mapper;
+    private final OpenAIClient openAIClient;
+
+    public IsaacLLMFreeTextValidator(final AbstractConfigLoader configLoader, final OpenAIClient openAIClient) {
+        this.openAIClient = openAIClient;
+        this.configLoader = configLoader;
+        this.mapper = new ObjectMapper();
+    }
+
+    private void validateInputs(final Question question, final Choice answer) {
+        // Validate question
+        Objects.requireNonNull(question);
+        if (!(question instanceof IsaacLLMFreeTextQuestion)) {
+            throw new IllegalArgumentException(question.getId() + " is not a LLM free-text question");
+        }
+
+        // Validate answer
+        Objects.requireNonNull(answer);
+        if (!(answer instanceof LLMFreeTextChoice)) {
+            throw new IllegalArgumentException(
+                    answer.getClass() + " is not of expected type FreeTextChoice for (" + question.getId() + ")");
+        }
+        int maxAnswerLength = 4096;
+        try { maxAnswerLength = Integer.parseInt(configLoader.getProperty(LLM_MARKER_MAX_ANSWER_LENGTH)); }
+        catch (NumberFormatException ignored) { /* Use default value */ }
+        if (answer.getValue().length() > maxAnswerLength) {
+            log.error("Answer exceeds maximum length for LLM free-text question marking: " + answer.getValue().length());
+            throw new IllegalArgumentException("Answer is too long for LLM free-text question marking");
+        }
+    }
+
+    private String generatePromptSystemMessage(final IsaacLLMFreeTextQuestion question) {
+        String contextAndInstructions = "You are a principal examiner marking A Level and GCSE "
+            + String.format("%s questions.\n", this.configLoader.getProperty(LLM_MARKER_SUBJECT))
+            + "You do not like vagueness in student answers.\n"
+            + "You follow a standard procedure when marking a question attempt and write your responses as JSON:\n"
+            + "```\n"
+            + "# QUESTION MARKING PROCEDURE\n"
+            + "if the attempt is similar enough to an attempt that you've marked before:\n"
+            + "  award the student the same json marks as you gave in that prior example.\n"
+            + "else:\n"
+            + "  for each mark in the mark scheme:\n"
+            + "    if the attempt meets the criteria of the mark: # taking into consideration any ADDITIONAL MARKING INSTRUCTIONS\n"
+            + "      record an entry in the JSON response indicating the field and the numeric mark value, i.e. `\"abbreviatedSnakeCaseMarkDescriptor\": 1`.\n"
+            + "\n"
+            + String.format("finally add a field `%s` to the JSON object ", MARKS_AWARDED_FIELD_NAME)
+            + "with the value equal to the sum of the earned marks while never exceeding the question's `maxMarks`.\n"
```

Comment:
Yes, I've also noticed this. It will require using an excel-like formula rather than sum because of some of the question marking logic but I have quite a clear idea of how to achieve this. In my plans for the next iteration. Thanks!